virtcontainers/kata_agent.go

```diff
@@ -1550,6 +1556,61 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) []*grpc.Storage
 	return epheStorages
 }
 
+// handleHugePages handles hugepages storage by
+// creating a Storage from corresponding source of the mount point
+func (k *kataAgent) handleHugepages(mounts []specs.Mount, hugepageLimits []specs.LinuxHugepageLimit) []*grpc.Storage {
+	//Map to hold the total memory of each type of hugepages
+	optionsMap := make(map[string]string)
+	var HugePageSizeUnitList = []string{"B", "K", "M", "G", "T", "P"}
+
+	for _, hp := range hugepageLimits {
+		if hp.Limit != 0 {
+			k.Logger().WithField("Pagesize", hp.Pagesize).Info("hugepage request")
+			k.Logger().WithField("Limit", hp.Limit).Info("hugepage request")
+			//example Pagesize 2MB, 1GB etc. The Limit are in Bytes
+			pageSize, _ := units.RAMInBytes(hp.Pagesize)
+			pageSizeStr := units.CustomSize("%g%s", float64(pageSize), 1024.0, HugePageSizeUnitList)
+			totalHpSize := fmt.Sprintf("%v", hp.Limit)
+			optionsMap[pageSizeStr] = totalHpSize
+		}
+	}
+
+	var hugepages []*grpc.Storage
+	for idx, mnt := range mounts {
+		//HugePages mount Type is Local
+		if mnt.Type == KataLocalDevType {
+			if _, fsType, fsOptions, _ := utils.GetDevicePathAndFsTypeOptions(mnt.Source); fsType == "hugetlbfs" {
+				k.Logger().WithField("fsOptions", fsOptions).Debug("hugepage mount options")
+				//Find the pagesize from the mountpoint to use the right set of options
+				//pagesize is the last element in the mount option. Can it change ?
+				pagesizeOpt := fsOptions[len(fsOptions)-1]
```

Comment:
Can you really count on the pagesize being the last option in the list?

Reply:
In my tests I always found it as the last option hence avoided searching the option. But I'm not sure hence I added a comment as well. Any ideas?

Reply:
@dgibson I looked at the kubernetes code and the kubernetes code only provides `pagesize` option - https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/emptydir/empty_dir.go#L381 which is then used for the actual hugepages mount on the host.
Checked the linux kernel (fs/hugetlbfs/inode.c) code as well. 
We should be able to safely assume that `pagesize` is the last option when it comes to Kubernetes. It doesn't use the `size` option. 
Let me know if you are ok, or want me to change the code to search for `pagesize` option ?

Reply:
Relying on exactly how external software (k8s in this case) presents the options is incredibly fragile.  You absolutely should search for the option.

Reply:
addressed this in latest iteration